components/post_deleted_modal.test.tsx

```diff
@@ -59,8 +59,8 @@ describe('components/ChannelInfoModal', () => {
                 onHide={emptyFunction}
             />
         );
-
-        const shouldUpdate = wrapper.instance().shouldComponentUpdate({show: true});
+        const instance = wrapper.instance();
+        const shouldUpdate = instance.shouldComponentUpdate && instance.shouldComponentUpdate({show: true}, {}, null);
```

Comment:
Can `shouldComponentUpdate` be null?

Reply:
It should never be null, but the type checking was failing and I wasn't sure of a more graceful solution for getting TS to play ball in this case (I've never had to call lifecycle methods from a test before)

Reply:
You can use the `!.` operator to have the TypeScript compiler dismiss the null check. If it should never be null, the app should throw an error if it is.

Reply:
I saw that there was a lint rule against using the `!` operator, so I wasn't sure if that was something we were avoiding, but I can do it and just disable the rule for this line

Reply:
I've removed that lint rule, you should be good to use it as long as your branch is sync'd with the latest master.

Reply:
Alright cool, thanks for letting me know. I've synced and made that update